### PR TITLE
Add storage method to delete time series

### PR DESCRIPTION
**Pull request by fabxc**
@beorn7 @juliusv

### CODE REVIEW

File: storage/local/storage_test.go [hunk 1]
```diff
@@ -204,6 +204,74 @@ func TestRetentionCutoff(t *testing.T) {
...
+
+	s, closer := NewTestStorage(t, 1)
+	defer closer.Close()
+
+	m1 := clientmodel.Metric{clientmodel.MetricNameLabel: "test", "n1": "v1"}
+	m2 := clientmodel.Metric{clientmodel.MetricNameLabel: "test", "n1": "v2"}
+
+	N := 120000
+
+	samples := make(clientmodel.Samples, N)
```
Comment (juliusv) [inline storage/local/storage_test.go]: This is never used except for being iterated over?

File: storage/local/storage_test.go [hunk 2]
```diff
@@ -204,6 +204,74 @@ func TestRetentionCutoff(t *testing.T) {
...
+		t.Fatalf("unexpected number of samples: %d", len(vals))
+	}
+	it = s.NewIterator(fps[1])
+	if vals := it.RangeValues(metric.Interval{insertStart, now}); len(vals) != N {
+		t.Fatalf("unexpected number of samples: %d", len(vals))
+	}
+
+	s.DropMetricsForFingerprints(fps[0])
+	s.WaitForIndexing()
+
```
Comment (juliusv) [inline storage/local/storage_test.go]: Maybe also re-check `FingerprintsForLabelMatchers()` to verify that the index deletion worked here and after the second drop.

File: storage/local/interface.go [hunk 3]
```diff
@@ -47,6 +47,9 @@ type Storage interface {
 	// The iterator will never return samples older than retention time,
 	// relative to the time NewIterator was called.
 	NewIterator(clientmodel.Fingerprint) SeriesIterator
+	// Drop all time series associated with the given fingerprints. This operation
+	// will not show up in the series operations metrics.
```
Comment (juliusv) [inline storage/local/interface.go]: Why not?

Comment (beorn7) [inline storage/local/interface.go]: The "purge" operation basically signals "a time series has not received any updated for the whole retention period and has therefore been deleted". If you manually delete time series, you are pretty aware that you did that. Showing those events in the same metric as the "real" purges is probably more confusing than helpful.

Comment (juliusv) [inline storage/local/interface.go]: Ok for now... if we ever have tools doing automated drops through this method, we should reconsider.

Comment (beorn7) [inline storage/local/interface.go]: In that case, we probably want a new series op type "manual_delete" or so...

File: storage/local/storage_test.go (hunk 1 above)
Comment (fabxc) [inline storage/local/storage_test.go]: Then `go` wouldn't compile it ;)
It's in the inner loop in line 221.

File: storage/local/storage_test.go (hunk 2 above)
Comment (fabxc) [inline storage/local/storage_test.go]: Good idea, thanks.

File: storage/local/storage_test.go (hunk 1 above)
Comment (beorn7) [inline storage/local/storage_test.go]: But that's what @juliusv means... Line 221 could be a plain counting loop.

Comment (juliusv) [inline storage/local/storage_test.go]: Exactly.

Comment (fabxc) [inline storage/local/storage_test.go]: I see, fixed.

File: storage/local/storage_test.go (hunk 2 above)
Comment (fabxc) [inline storage/local/storage_test.go]: Done.

**Comment by beorn7** [comment]
:+1:

**Comment by juliusv** [comment]
:+1:

**Pull request merged**